bambi/backend/pymc.py

```diff
@@ -203,43 +221,27 @@ def _run_mcmc(
                             random_seed=random_seed,
                             **kwargs,
                         )
+                        idata_from = "pymc"
                     else:
                         raise
-            elif sampler_backend == "nuts_numpyro":
-                import pymc.sampling_jax  # pylint: disable=import-outside-toplevel
-
-                if not chains:
-                    # sample_numpyro_nuts does not handle chains = None like pm.sample does
-                    chains = 4
-                idata = pymc.sampling_jax.sample_numpyro_nuts(
-                    draws=draws,
-                    tune=tune,
-                    chains=chains,
-                    random_seed=random_seed,
-                    **kwargs,
-                )
-            elif sampler_backend == "nuts_blackjax":
-                import pymc.sampling_jax  # pylint: disable=import-outside-toplevel
-
-                # sample_blackjax_nuts does not handle chains = None like pm.sample does
-                if not chains:
-                    chains = 4
-                idata = pymc.sampling_jax.sample_blackjax_nuts(
-                    draws=draws,
-                    tune=tune,
-                    chains=chains,
-                    random_seed=random_seed,
-                    **kwargs,
-                )
-            else:
-                raise ValueError(
-                    f"sampler_backend value {sampler_backend} is not valid. Please choose one of"
-                    f"'mcmc', 'nuts_numpyro' or 'nuts_blackjax'"
-                )
-        idata = self._clean_results(idata, omit_offsets, include_mean)
+        elif sampler_backend in BAYEUX_SAMPLERS:
+            import bayeux as bx
+            import jax
+
+            bx_model = bx.Model.from_pymc(self.model)
```

Comment:
ok, as of bayeux 0.1.9, you could use `bx_model.methods`, which returns a dictionary. I would do something like

```
def maybe_get_bayeux_method(sampler_backend):
  if sampler_backend.count('.') == 1:
    return sampler_backend.split('.')
  return None

bayeux_method = maybe_get_bayeux_method(sampler_backend)

...later

if bayeux_method is not None:
  inference_kind, inference_type = bayeux_method
  bx_model = bx.Model.from_pymc(self.model)
  if inference_kind in bx_model.methods and inference_type in bx_model.methods[inference_kind]:
    sampler = getattr(getattr(bx_model, inference_kind), inference_type)
    res = sampler(seed=seed, **kwargs)
    if inference_kind == "mcmc":
      res = self._clean_results(res, ...)
    return res
formatted_methods = [f'{k}.{v} for k, vals in bx_model.methods.items() for v in vals]
raise ValueError(f"Use one of {formatted_methods}")
```

Reply:
Is this suggesting that users pass, for example, `mcmc.blackjax_nuts`, `mcmc.numpyro_nuts` instead of `blackjack_nuts`, `numpyro_nuts` to the parameter `inference_method`?

Reply:
yeah, exactly. and it allows optimization or vi to be run (but then the return type is kind of a wildcard)